presto-hive/src/test/java/com/facebook/presto/hive/TestHiveNativeWorkersQueries.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive;
+
+import com.facebook.presto.testing.QueryRunner;
+import com.facebook.presto.tests.AbstractTestQueryFramework;
+import com.facebook.presto.tests.DistributedQueryRunner;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Optional;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static io.airlift.tpch.TpchTable.NATION;
+
+public class TestHiveNativeWorkersQueries
```

Comment:
It would be nice to have a default which just loads up Presto Java workers.  That would ensure this test actually works, and potentially spot and document discrepancies between Java vs. custom workers.

Reply:
I'm not sure I understand this suggestion. The queries are very simple and there is not much benefit in running them against Java workers. The verification is happening using H2 just like any other test, hence, there should not be any concern about the test's correctness. Am I missing something?

Reply:
Hmm, I missed this, how is verification happening with H2?  My only point was, if we don't want to introduce a dependency from PrestoDB (Java source) to C++ native worker source which exists somewhere else, but still want to make sure that future tests are sane, it would be nice to have a default, and an easy way to accomplish that would have the default be a vanilla Java worker to serve as a baseline.  Not sure how H2 verification is happening in this test.

Reply:
`assertQuery("SELECT * FROM nation");` runs the query using queryRunner (native workers) and H2, then compares the results.